changelogs/unreleased/gh-7166-deprecate-fiber_set_cancellable.md

```diff
@@ -0,0 +1,3 @@
+## feature/core
+
+* **[Breaking change]** `fiber_set_cancellable()` is deprecated (gh-7166).
```

Comment:
From my point of view 'deprecated' usually means that nothing changes in a behaviour (except, maybe, a warning), but the usage of given functioonality is discouraged. If we'll change it to 'deprecated and now does nothing' or something of this kind, it would be clear what is going on.

Reply:
Done